packages/xo-web/src/xo-app/home/backed-up-vms.js

```diff
@@ -0,0 +1,248 @@
+          <Link to={`/${container.type}s/${container.id}`}>
+            {renderXoItem(container)}
+          </Link>
+        )
+      )
+    },
+  },
+]
+
+const ACTIONS = [
+  {
+    handler: vms => {
+      if (every(vms, _ => _.power_state === 'Halted')) {
+        return startVms(vms)
+      }
+
+      return stopVms(vms)
+    },
+    icon: vms => {
+      if (every(vms, _ => _.power_state === 'Halted')) {
+        return 'vm-start'
+      }
+
+      return 'vm-stop'
+    },
+    label: vms => {
+      if (every(vms, _ => _.power_state === 'Halted')) {
+        return _('startVmLabel')
+      }
+
+      return _('stopVmLabel')
+    },
+  },
+  {
+    handler: migrateVms,
+    icon: 'vm-migrate',
+    label: _('migrateVmLabel'),
+  },
+  {
+    handler: snapshotVms,
+    icon: 'vm-snapshot',
+    label: _('snapshotVmLabel'),
+  },
+  {
+    handler: copyVms,
+    icon: 'vm-copy',
+    label: _('copyVmLabel'),
+  },
+  {
+    handler: (vms, { setHomeVmIdsSelection }) => {
+      setHomeVmIdsSelection(map(vms, 'id'))
+    },
+    icon: 'backup',
+    label: _('backupLabel'),
+    redirectOnSuccess: '/backup/new/vms',
+  },
+  {
+    handler: deleteVms,
+    icon: 'vm-delete',
+    label: _('vmRemoveButton'),
+    level: 'danger',
+  },
+]
+
+const GROUPED_ACTIONS = [
+  {
+    disabled: vms => some(vms, _ => _.power_state !== 'Running'),
+    handler: restartVms,
+    icon: 'vm-reboot',
+    label: _('rebootVmLabel'),
+  },
+  {
+    disabled: vms => some(vms, _ => _.power_state !== 'Running'),
+    handler: pauseVms,
+    icon: 'vm-pause',
+    label: _('pauseVmLabel'),
+  },
+  {
+    disabled: vms => some(vms, _ => _.power_state !== 'Running'),
+    handler: suspendVms,
+    icon: 'vm-suspend',
+    label: _('suspendVmLabel'),
+  },
+]
+
+export const NotBackedUpVms = decorate([
+  addSubscriptions({
+    jobs: subscribeBackupNgJobs,
+  }),
+  connectStore(() => ({
+    hosts: createGetObjectsOfType('host'),
+    pools: createGetObjectsOfType('pool'),
+  })),
+  provideState({
+    computed: {
+      notBackedUpVms: (_, { jobs, vms }) => {
+        if (isEmpty(vms)) {
+          return []
+        }
+
+        const backedUpVms = map(jobs, job =>
+          find(vms, createPredicate(omit(job.vms, 'power_state')))
+        )
+        return vms.filter(vm => !backedUpVms.includes(vm))
+      },
+    },
+  }),
+  injectState,
+  ({ hosts, pools, setHomeVmIdsSelection, state: { notBackedUpVms } }) => (
+    <div>
+      <h5>{_('notBackedUpVms')}</h5>
+      <SortedTable
+        actions={ACTIONS}
+        collection={notBackedUpVms}
+        columns={COLUMNS}
+        data-hosts={hosts}
+        data-pools={pools}
+        data-setHomeVmIdsSelection={setHomeVmIdsSelection}
+        groupedActions={GROUPED_ACTIONS}
+        shortcutsTarget='body'
+        stateUrlParam='s'
+      />
+    </div>
+  ),
+])
+
+NotBackedUpVms.propTypes = {
+  vms: PropTypes.arrayOf(PropTypes.object),
+}
+
+const BackedUpVms = decorate([
```

Comment:
Why do you need 2 separate components? Can't you just pass a prop?

Reply:
Done